### ISSUE TITLE
Chrome / desktop apps sample?

### ISSUE THREAD

**Issue by orlin** [binaryage/chromex#2]
Could Chromex + Dirac be the best possible way to develop desktop apps with CLJS right now?  People may think that's Electron, but why are the [react-native tools](http://cljsrn.org/) so much far better?  Perhaps people are waiting for react-native to make its way to the desktop, which is already on its way for [Mac OS](https://github.com/ptmt/react-native-desktop) + maybe also [Linux](https://github.com/ptmt/react-native-desktop/issues/59), and [Windows](http://www.reactnative.com/react-native-on-the-universal-windows-platform/) for sure.  Not many can afford to write the same thing for every platform though.  For me web ui on desktop is just fine.  Since the recent [v0.13 of NW.js](http://nwjs.io/blog/whats-new-in-0.13/) Chrome Apps can be packaged to install without Chrome's web store.  Electron on the other hand [cannot work on ChromeOS](https://discuss.atom.io/t/chrome-apps-support/19975) as it stands.

Do I understand correctly that you wrote Chromex so you could write Dirac so you could write web apps for the web?  Would it be too much to ask for a chromex-app-sample?  The #readme says "the same patterns generally apply to Chrome App development as well", but how would the template differ for apps?  Ideally using Dirac too.  The only trace of dirac I see in chromex-sample is [.figwheel_dirac.log](https://github.com/binaryage/chromex-sample/blob/master/project.clj#L69), guessing it's a leftover from Dirac itself.  Ideally I would reuse the same project for the website and share components with the chromex app.  Furthermore, an app could maybe run in either desktop or web context, conditionally using chrome-dependent components (which won't be many in my case) and demanding login for web use.  My clojure experience is kind of thin so any tips would be appreciated.

**Comment by darwin** [comment]
Well, one of my motivations for writing Chromex was to have ability to write Chrome extensions and later desktop apps on top of Chrome platform (with ChromeOS support). But the immediate main reason was a closed-source Chrome extension I have in development. Writing Dirac using Chromex was just a consequence because Chromex happened exists. However, AFAIK the Dirac is currently the largest open-source codebase using Chromex.

> Would it be too much to ask for a chromex-app-sample?

I'm sorry, I don't work on any chrome app currently, so I'm not going to build it at this point.

Chromex should have no knowledge about Dirac. Using Dirac should be independent. Dirac should work with any platform which runs on top of Chrome and allows specifying remote debugging port (for [Dirac DevTools to connect to](https://github.com/binaryage/dirac/blob/master/docs/faq.md#why-should-i-use-recent-chrome-canary-with-dirac-devtools)). According to my experience, this is possible with normal Chrome binaries (for developing web apps, chrome extensions, chrome apps) and even with Electron where you can launch Dirac DevTools independently and connect it to Chrome backend powered by Electron.

Long story short: I have never built any Chrome App using Chromex, but I strongly believe that the project template would be almost the same and same patterns would apply. I have this confidence because generated core APIs are mostly 1:1 between extensions and apps. The developer should not feel much difference except for list of capabilities available. And this is I believe by design. Chrome authors don't want web developers to learn anything new to develop Chrome Apps.

**Comment by darwin** [comment]
Chrome Apps are dead in the water:
http://blog.chromium.org/2016/08/from-chrome-apps-to-web.html

I will eventually remove support for them from Chromex.
